coil-base/src/main/java/coil/fetch/FileFetcher.kt

```diff
@@ -5,13 +5,18 @@ import coil.ImageLoader
 import coil.decode.DataSource
 import coil.decode.ImageSource
 import coil.request.Options
+import coil.util.extension
+import okio.FileSystem
+import okio.Path
+import okio.Path.Companion.toOkioPath
 import java.io.File
 
-internal class FileFetcher(private val data: File) : Fetcher {
+internal class FileFetcher(private val data: Path, private val fileSystem: FileSystem) : Fetcher {
```

Comment:
Let's keep the `data: File` constructor only; it's convention for `Fetcher`s to perform any data mapping inside `fetch`.